server/src/main/java/org/opentosca/toscana/core/plugin/TOSCAnaPlugin.java

```diff
@@ -1,36 +1,45 @@
-package org.opentosca.toscana.plugins.lifecycle;
+package org.opentosca.toscana.core.plugin;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
-import org.opentosca.toscana.core.plugin.AbstractPlugin;
+import org.opentosca.toscana.core.plugin.lifecycle.ExecutionPhase;
+import org.opentosca.toscana.core.plugin.lifecycle.TransformationLifecycle;
+import org.opentosca.toscana.core.plugin.lifecycle.ValidationFailureException;
 import org.opentosca.toscana.core.transformation.TransformationContext;
 import org.opentosca.toscana.core.transformation.platform.Platform;
 
 import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  This class represents the base class for every Plugin that wants to implement the Transformation lifecycle
 
  @param <LifecycleT> The class type of the plugin specific implementation of the Lifecycle interface */
-public abstract class LifecycleAwarePlugin<LifecycleT extends TransformationLifecycle>
-    extends AbstractPlugin {
+public abstract class TOSCAnaPlugin<LifecycleT extends TransformationLifecycle> {
 
     /**
      Immutable list of execution tasks, they are in the right execution order and get executed from the first
      index to the last
      */
     private final List<ExecutionPhase<LifecycleT>> executionTasks;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Platform platform;
 
     /**
      Initializes the plugin, that means that all tasks get added to a internal list (executionTasks) that then gets
      executed by the transform method.
      <p>
      The tasks get stored in a list (environment tasks)
      */
-    public LifecycleAwarePlugin(Platform platform) {
-        super(platform);
+    public TOSCAnaPlugin(Platform platform) {
+        this.platform = platform;
+        if (platform == null) {
+            throw new IllegalStateException("Platform must not be null");
```

Comment:
Could be simplified like this:
this.platform = Objects.requireNonNull(platform);